web3/eth/base_eth.py

```diff
@@ -95,17 +95,28 @@ def set_gas_price_strategy(
         self._gas_price_strategy = gas_price_strategy
 
     def estimate_gas_munger(
-        self, transaction: TxParams, block_identifier: Optional[BlockIdentifier] = None
-    ) -> Sequence[Union[TxParams, BlockIdentifier]]:
+        self,
+        transaction: TxParams,
+        block_identifier: Optional[BlockIdentifier] = None,
+        state_override: Optional[CallOverride] = None,
+    ) -> Sequence[Union[TxParams, BlockIdentifier, CallOverride]]:
         if "from" not in transaction and is_checksum_address(self.default_account):
             transaction = assoc(transaction, "from", self.default_account)
 
         if block_identifier is None:
-            params: Sequence[Union[TxParams, BlockIdentifier]] = [transaction]
-        else:
-            params = [transaction, block_identifier]
+            return [transaction]
 
-        return params
+        if state_override is None:
+            return [
+                transaction,
+                block_identifier,
```

Comment:
This is a nit but it's really just nice to make this more concise. If you remove the comma after the last item in a list of dict, and it fits into the line without triggering a line-length violation, it will look like this after `black` runs:

```py
return [transaction, block_identifier]
```

The same can be done by removing the trailing comma after `state_override` below. It makes the code nicer / easier to read.